pingpong/ud_pingpong.c

```
@@ -519,7 +519,11 @@ static int run(void)
 		for (i = 0; i < TEST_CNT; i++) {
 			if (test_size[i].option > opts.size_option)
 				continue;
+
 			opts.transfer_size = test_size[i].size;
+			if (opts.transfer_size > buffer_size)
+				continue;
```

> If I understand you correctly you are recommending that when the -S option
> is given I give the value as an ep hint and the provider can return -
> FI_ENODATA if it is too high. Is that right?

yep

> If so, what should be the behavior for the default test cases? (no error
> printed, and skip the ones that are too large?)

I would silently skip them.